### PR TITLE
pass-url-arg-1

### LINKED ISSUE #1: Pass the page URL as an arg to the thing engine
Add something like URL=http...  to the URI we use to go to the thing engine. That will give it more context to use in deciding what to do.

### PR DESCRIPTION
**Pull request by terrycojones**
jkakar:approve ntoll: approve

Always pass a url=xxx arg giving the URL of the page the user was on when they used the extension.

Fixes #1

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -31,60 +32,97 @@ chrome.pageAction.onClicked.addListener(
...
+
+function getClickHandlerPage() {
+  return function(info, tab){
+    makeTab(info.pageUrl, info, tab);
+  };
+}
+
 chrome.contextMenus.create({
-    'title' : 'View selected text in ' + product,
+    'title' : 'View page URL in ' + product,
```
Comment (ntoll) [inline background.js]: Very trivial, but I'd have avoided using a technical term like "URL". How about, "View target link in "...?

Comment (terrycojones) [inline background.js]: This one is not for a link. It's what you see if you do a right click on the page background (i.e., not over a link or an image or with text selected). I could say "view link for this page" but it sounds vague to me. Or maybe "View link to this page"?

File: background.js [hunk 2]
```diff
@@ -31,60 +32,97 @@ chrome.pageAction.onClicked.addListener(
...
+  var ref = referer(info);
+  if (ref === undefined){
+    return base + encodeURIComponent(about);
+  }
+  else {
+    return base + encodeURIComponent(about) + '?' + ref;
+  }
+}
+
+function referer(info){
```
Comment (jkakar) [inline background.js]: The indentation of this function doesn't match the rest of the file.

File: background.js [hunk 3]
```diff
@@ -31,60 +32,97 @@ chrome.pageAction.onClicked.addListener(
   }
 );
 
-// --------------------------- Selection, Link, Page --------------------------
-
-function getClickHandlerSelectionLinkOrPage() {
+// ----------------- Utility functions for context menus -----------------
+
+function makeTab(about, info, tab){
```
Comment (jkakar) [inline background.js]: The would be a touch more clear/obvious as `openNewTab`.

File: background.js [hunk 4]
```diff
@@ -31,60 +32,97 @@ chrome.pageAction.onClicked.addListener(
...
+   * A utility function to produce a url=xxx refering page URL fragment for a
+   * request to the object browser.
+   */
+    if (info.pageUrl){
+      return 'url=' + encodeURIComponent(info.pageUrl);
+    }
+    else {
+      return '';
+    }
+}
```
Comment (jkakar) [inline background.js:67]: You might consider shortening this:

```
return info.pageUrl ? 'url=' + encodeURIComponent(info.pageUrl) : '';
```

File: background.js (hunk 3 above)
Comment (terrycojones) [inline background.js]: Done.

File: background.js (hunk 2 above)
Comment (terrycojones) [inline background.js]: Thanks, fixed.

File: background.js (hunk 4 above)
Comment (terrycojones) [inline background.js:67]: Done, thx.

**Pull request merged**